engine/crates/federated-graph/src/render_sdl/render_api_sdl.rs

```diff
@@ -0,0 +1,246 @@
+            f.write_str(&graph[input_object.name])?;
+            write_public_directives(f, input_object.composed_directives, graph)?;
+
+            f.write_char(' ')?;
+
+            write_block(f, |f| {
+                for field in &graph[input_object.fields] {
+                    if has_inaccessible(&field.directives, graph) {
+                        continue;
+                    }
+
+                    write_description(f, field.description, INDENT, graph)?;
+                    let field_name = &graph[field.name];
+                    f.write_str(INDENT)?;
+                    f.write_str(field_name)?;
+                    f.write_str(": ")?;
+                    f.write_str(&render_field_type(&field.r#type, graph))?;
+                    write_public_directives(f, field.directives, graph)?;
+                    f.write_char('\n')?;
+                }
+
+                Ok(())
+            })?;
+
+            f.write_char('\n')?;
+        }
+
+        for union in &graph.unions {
+            if has_inaccessible(&union.composed_directives, graph) {
+                continue;
+            }
+
+            write_leading_whitespace(f)?;
+
+            write_description(f, union.description, "", graph)?;
+            f.write_str("union ")?;
+            f.write_str(&graph[union.name])?;
+            write_public_directives(f, union.composed_directives, graph)?;
+            f.write_str(" =")?;
+
+            let mut members = union.members.iter().peekable();
+
+            while let Some(member) = members.next() {
+                f.write_str(" ")?;
+                f.write_str(&graph[graph[*member].name])?;
+
+                if members.peek().is_some() {
+                    f.write_str(" |")?;
+                }
+            }
+
+            f.write_char('\n')?;
+        }
+
+        for scalar in &graph.scalars {
+            let scalar_name = &graph[scalar.name];
+
+            if BUILTIN_SCALARS.contains(&scalar_name.as_str()) || has_inaccessible(&scalar.composed_directives, graph) {
+                continue;
+            }
+
+            write_leading_whitespace(f)?;
+
+            write_description(f, scalar.description, "", graph)?;
+            f.write_str("scalar ")?;
+            f.write_str(scalar_name)?;
+            write_public_directives(f, scalar.composed_directives, graph)?;
+
+            f.write_char('\n')?;
+        }
+
+        Ok(())
+    }
+}
+
+fn has_inaccessible(directives: &Directives, graph: &FederatedGraphV3) -> bool {
+    graph[*directives]
+        .iter()
+        .any(|directive| matches!(directive, Directive::Inaccessible))
+}
+
+fn write_public_directives(
+    f: &mut fmt::Formatter<'_>,
+    directives: Directives,
+    graph: &FederatedGraphV3,
+) -> fmt::Result {
+    for directive in graph[directives].iter().filter(|directive| match directive {
+        Directive::Authenticated | Directive::Inaccessible | Directive::Policy(_) | Directive::RequiresScopes(_) => {
```

Comment:
as mentioned, would include `@authenticated`, `@requiresScopes` and `@requires`. Probably wouldn't bother with `@policy` though, we can't use it...